Gemfile

```diff
@@ -9,12 +9,18 @@ solidus_branch = ENV.fetch('SOLIDUS_BRANCH', 'master')
 gem 'solidus', github: 'solidusio/solidus', branch: solidus_branch
 gem 'solidus_auth_devise'
 
-if ENV['DB'] == 'mysql'
+case ENV['DB']
+when 'mysql'
   gem 'mysql2', '~> 0.4.10'
-else
+when 'postgres'
   gem 'pg', '~> 0.21'
 end
 
+case ENV['LAZY_RESOLVER']
```

Comment:
@kennyadsl exactly, the `batch-loader` gem can't be added to the `gemspec` because we use the Adapter pattern; it's like if ActiveRecord would depend from sqlite, pg, mysql2 and the other adapters that it implements. We'll just need to document that in order to use this extension you'll have to add `batch-loader` to the Gemfile. About the env var: I use it for Travis, I just copied the approach used for databases (`case ENV['DB']; when 'mysql' then gem 'mysql2'; when 'postgres' then gem 'pg'; etc`), do you know a better approach?

Reply:
I'm just worried that people will not read the README and a default would probably make life easier for them. 

Can't we conditionally add this gem in the `gemspec` as well, as we are doing for the `Gemfile`?

Reply:
AFAIK we can't; anyway it's just the Adapter pattern, like ActiveRecord, ActiveJob, it's something well known, it shouldn't be so surprising; also we have the dependency check in the Adapter: https://github.com/solidusio-contrib/solidus_graphql_api/pull/65/files#diff-82c6c00a0cc5f75a48716c0402642a76R14 so if you don't have that gem and you are using that Adapter it will tell you to install it

Reply:
The alternative is not to use the Adapter pattern and just stick with `batch-loader`, but it's not very attractive to me

Reply:
What about an option in the install command that appends `batch-loader` by default (unless you specify other options) to your Gemfile, similar to what is done by Rails when you create a new app?

Anyway, we can even improve later. I don't want to block development for this, but can we add a section to the README about this in this same PR?

Reply:
@kennyadsl I added generator logic in order to install the lazy resolvers adapter dependency (and also updated the README), solving the problem in the generator phase seems to me the best approach too :+1: